hudi-hadoop-common/src/main/java/org/apache/hudi/io/hadoop/HoodieHBaseAvroHFileReader.java

```diff
@@ -370,10 +370,7 @@ public IndexedRecord next() {
     return new KeyPrefixIterator();
   }
 
-  private static Iterator<IndexedRecord> getRecordByKeyIteratorInternal(HFileScanner scanner,
-                                                                        String key,
-                                                                        Schema writerSchema,
-                                                                        Schema readerSchema) throws IOException {
+  private static Option<IndexedRecord> fetchRecordByKeyInternal(HFileScanner scanner, String key, Schema writerSchema, Schema readerSchema) throws IOException {
     KeyValue kv = new KeyValue(getUTF8Bytes(key), null, null, null);
```

Comment:
`getUTF8Bytes(key)` is evaluated twice, here and at line 397. Let's move this into a variable to ensure it's only computed once.